content/concepts/introduction/protocols/dht.md

```diff
@@ -1,9 +1,99 @@
+[Kademlia whitepaper](https://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf)
+and is augmented with other systems, such as
+[S/Kademlia](https://attachment.victorlampcdn.com/article/content/20220705/SKademlia_2007.pdf),
+[Coral](https://www.cs.princeton.edu/~mfreed/docs/coral-iptps03.pdf), and the
+[BitTorrent DHT](https://www.bittorrent.org/beps/bep_0005.html).
+
+Kad-DHT offers a way to find and manage nodes and data on the network by using a
+[routing table](https://en.wikipedia.org/wiki/Routing_table) that organizes peers based
+on how similar their keys are.
+
+<details>
+  <summary>A look deeper</summary>
+
+  The routing table is organized based on a prefix length and a distance metric.
+  The prefix length helps to group similar keys, and the distance metric helps to
+  find the closest peers to a specific key in the routing table. The table maintains
+  a list of `k` closest peers for each possible prefix length between `0` and `L-1`,
+  where `L` is the length of the keyspace, determined by the length of the hash
+  function used. **Kad-DHT uses SHA-256**, with a keyspace of 256 bits, maintaining
+  `k` peers with a shared key prefix for every prefix length between `0` and `255` in
+  its routing table.
+
+  The prefix length measures the proximity of two keys in the routing table and
+  divides the keyspace into smaller subspaces, called "buckets", each containing nodes
+  that share a common prefix of bits in their SHA-256 hash. The prefix length is the
+  number of bits that are the same in the two keys' SHA-256 hash. The more leading bits
+  that are the same, the shorter the prefix length and the closer the proximity of the
+  two keys are considered to be.
+
+  The distance metric is a way to calculate the distance between two keys by taking
+  the bitwise exclusive-or (XOR) of the SHA-256 hash of the two keys. The resulting
+  hash is a measure of the distance between the two keys, where a distance of `0` means
+  the keys are identical, and a distance of `1` means that only one bit is different,
+  meaning the two keys are close to each other (i.e. their SHA-256 hashes are similar).
+
+  This allows for efficient and effective lookups in the routing table when trying
+  to find nodes or data that share similar prefixes.
+
+</details>
+
+## Peer discovery
+
+The Kad-DHT uses a process called "peer routing" to discover new nodes in the network.
+This process starts by generating a random peer ID and looking it up via the routing
+table. The node then contacts the k closest nodes to the peer ID and repeats the process
+until it finds the peer or determines that it is not in the network. Nodes also adds any
+new nodes they discovers to their routing table to improve its awareness of the network.
+
+This process can also find nodes with a specific key prefix. Check out the Kad-DHT
+discovery document for more information.
+
+<!-- to add link -->
+
+## Content provider discovery
```

Comment:
Worth mentioning key/value (the high level APIs behind `GET_VALUE` and `PUT_VALUE`) here?